WordPress/src/test/java/org/wordpress/android/ui/reader/repository/usecases/PostLikeUseCaseTest.kt

```diff
@@ -31,60 +48,273 @@ class PostLikeUseCaseTest {
+            testWithUnlikedPost {
+                // Act
+                val result = useCase.perform(readerPost, true).toList(mutableListOf())
+
+                // Assert
+                assertThat((result)).contains(Success)
+            }
+
+    @Test
+    fun `success is returned when unliking a like post`() =
+            testWithLikedPost {
+                // Act
+                val result = useCase.perform(readerPost, false).toList(mutableListOf())
+
+                // Assert
+                assertThat((result)).contains(Success)
+            }
+
+    @Test
+    fun `failure is returned when liking an unliked post`() =
+            testWitFailurehUnlikedPost {
+                // Act
+                val result = useCase.perform(readerPost, true).toList(mutableListOf())
+
+                // Assert
+                assertThat((result)).contains(Failed.RequestFailed)
+            }
+
+    @Test
+    fun `failure is returned when unliking a like post`() =
+            testWithFailureLikedPost {
+                // Act
+                val result = useCase.perform(readerPost, false).toList(mutableListOf())
+
+                // Assert
+                assertThat((result)).contains(Failed.RequestFailed)
+            }
+
+    @Test
+    fun `like local action is triggered for selected reader post`() =
+            testWithUnlikedPost {
+                // Act
+                useCase.perform(readerPost, true).toList(mutableListOf())
+
+                // Assert
+                verify(readerPostActionsWrapper).performLikeActionLocal(
+                        anyOrNull(),
+                        anyBoolean(),
+                        anyLong()
+                )
+            }
+
+    @Test
+    fun `like remote action is triggered for selected reader post`() =
+            testWithUnlikedPost {
+                // Act
+                useCase.perform(readerPost, true).toList(mutableListOf())
+
+                // Assert
+                verify(readerPostActionsWrapper).performLikeActionRemote(
+                        anyOrNull(),
+                        anyBoolean(),
+                        anyLong(),
+                        anyOrNull()
+                )
+            }
+
+    @Test
+    fun `Post views bumped when asking to like`() =
+            testWithUnlikedPost {
+                // Act
+                useCase.perform(readerPost, true).toList(mutableListOf())
+
+                // Assert
+                verify(readerPostActionsWrapper).bumpPageViewForPost(anyOrNull())
+            }
+
+    @Test
+    fun `Post views NOT bumped when asking to unlike`() =
+            testWithLikedPost {
+                // Act
+                useCase.perform(readerPost, false).toList(mutableListOf())
+
+                // Assert
+                verify(readerPostActionsWrapper, never()).bumpPageViewForPost(anyOrNull())
+            }
+
+    private fun <T> testWithUnlikedPost(block: suspend CoroutineScope.() -> T) {
```

Comment:
We might want to consider replacing all these methods with 

```
private fun init(
        isLikedByCurrentUser: Boolean = false,
        isNetworkAvailable: Boolean = true,
        succeeds: Boolean = true
    ) {
        readerPost.isLikedByCurrentUser = isLikedByCurrentUser
        whenever(networkUtilsWrapper.isNetworkAvailable()).thenReturn(isNetworkAvailable)
        whenever(readerPostActionsWrapper.performLikeActionLocal(anyOrNull(), anyBoolean(), anyLong()))
                .thenReturn(true)
        whenever(readerPostActionsWrapper.performLikeActionRemote(anyOrNull(), anyBoolean(), anyLong(), anyOrNull()))
                .then {
                    (it.arguments[3] as ActionListener).onActionResult(succeeds)
                }
    }
```
We'd call 
`init(isLikedByCurrentUser = false)` for `testWithUnlikedPost`
`init(isLikedByCurrentUser = false, succeeds = false)` for `testWitFailurehUnlikedPost`
etc.

Wdyt?

Reply:
Sure, sounds good. In commit 9a6f6e8